vignettes/predict_moa_using_snf.Rmd

```diff
@@ -0,0 +1,443 @@
+---
+title: "Predict compounds mechanism-of-action by morphological profiling"
+author: "Allen Goodman and Shantanu Singh"
+date: "`r Sys.Date()`"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Predict compounds mechanism-of-action by morphological profiling}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+This vignette demonstrates predicting compound mechanism-of-action using
+morphological profiling data. The data is "cleaned up" using the similarity 
+network fusion algorithm (SNFD). See the vignette `single_cell_analysis` for
+details about this dataset.
+
+```{r libraries, message=FALSE}
+library(dplyr)
+library(magrittr)
+library(ggplot2)
+library(stringr)
+library(cytominergallery)
+library(SNFtool)
+```
+
+# Load data
+
+Per-well profiles computed in `single_cell_analysis` are loaded, as well as
+metadata associated with these profiles (obtained from [BBBC021](https://data.broadinstitute.org/bbbc/BBBC021/))
+
+```{r message=FALSE}
+profiles <-
+  readr::read_csv(system.file("extdata", "ljosa_jbiomolscreen_2013_per_well_mean.csv",
+                package = "cytominergallery"))
+
+moa <-
+  readr::read_csv(system.file("extdata", "BBBC021_v1_moa.csv",
+                              package = "cytominergallery")) %>%
+  rename(Image_Metadata_Compound = compound,
+                Image_Metadata_Concentration = concentration,
+                Image_Metadata_MoA = moa
+  )
+
+metadata <-
+  readr::read_csv(system.file("extdata", "BBBC021_v1_image.csv",
+                              package = "cytominergallery")) %>%
+  rename(Image_Metadata_Plate = Image_Metadata_Plate_DAPI,
+                Image_Metadata_Well = Image_Metadata_Well_DAPI
+  ) %>%
+  select(matches("^Image_Metadata")) %>%
+  inner_join(moa) %>%
+  distinct()
+
+profiles %<>%
+  inner_join(metadata)
+
+variables <-
+  colnames(profiles) %>%
+  str_subset("^Nuclei_|^Cells_|^Cytoplasm_")
+
+```
+
+How many compounds?
+
+```{r}
+profiles %>%
+  filter(Image_Metadata_Compound != "DMSO") %>%
+  distinct(Image_Metadata_Compound) %>%
+  tally() %>%
+  rename(`Number of compounds` = n) %>%
+  knitr::kable()
+```
+
+How many unique treatments (compound-concentration pairs)?
```

Comment:
Remove chunk because it is the same as in predict_moa and doesn't present anything new